pkg/kubelet/config/http_test.go

```diff
@@ -281,3 +282,192 @@ func TestExtractFromHTTP(t *testing.T) {
+					},
+					Spec: api.PodSpec{
+						RestartPolicy: api.RestartPolicyAlways,
+						DNSPolicy:     api.DNSClusterFirst,
+						Containers: []api.Container{{
+							Name:  "1",
+							Image: "foo",
+							TerminationMessagePath: "/dev/termination-log",
+							ImagePullPolicy:        "Always"}},
+					},
+				}),
+		},
+		{
+			desc: "Multiple pods",
+			pods: v1beta3.PodList{
+				TypeMeta: v1beta3.TypeMeta{
+					Kind:       "PodList",
+					APIVersion: "v1beta3",
+				},
+				Items: []v1beta3.Pod{
+					{
+						ObjectMeta: v1beta3.ObjectMeta{
+							Name: "foo",
+							UID:  "111",
+						},
+						Spec: v1beta3.PodSpec{
+							Containers: []v1beta3.Container{{Name: "1", Image: "foo", ImagePullPolicy: v1beta3.PullAlways}},
+						},
+					},
+					{
+						ObjectMeta: v1beta3.ObjectMeta{
+							Name: "bar",
+							UID:  "222",
+						},
+						Spec: v1beta3.PodSpec{
+							Containers: []v1beta3.Container{{Name: "2", Image: "bar", ImagePullPolicy: ""}},
+						},
+					},
+				},
+			},
+			expected: CreatePodUpdate(kubelet.SET,
+				kubelet.HTTPSource,
+				api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						UID:       "111",
+						Name:      "foo" + "-" + hostname,
+						Namespace: "foobar",
+						SelfLink:  "/api/v1beta2/pods/foo-" + hostname + "?namespace=default",
+					},
+					Spec: api.PodSpec{
+						RestartPolicy: api.RestartPolicyAlways,
+						DNSPolicy:     api.DNSClusterFirst,
+						Containers: []api.Container{{
+							Name:  "1",
+							Image: "foo",
+							TerminationMessagePath: "/dev/termination-log",
+							ImagePullPolicy:        "Always"}},
+					},
+				},
+				api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						UID:       "222",
+						Name:      "bar" + "-" + hostname,
+						Namespace: "foobar",
+						SelfLink:  "/api/v1beta2/pods/bar-" + hostname + "?namespace=default",
+					},
+					Spec: api.PodSpec{
+						RestartPolicy: api.RestartPolicyAlways,
+						DNSPolicy:     api.DNSClusterFirst,
+						Containers: []api.Container{{
+							Name:  "2",
+							Image: "bar",
+							TerminationMessagePath: "/dev/termination-log",
+							ImagePullPolicy:        "IfNotPresent"}},
+					},
+				}),
+		},
+		{
+			desc:     "Empty Array",
+			pods:     []v1beta3.Pod{},
+			expected: CreatePodUpdate(kubelet.SET, kubelet.HTTPSource),
+		},
+	}
+
+	for _, testCase := range testCases {
+		data, err := json.Marshal(testCase.pods)
+		if err != nil {
+			t.Fatalf("%s: Some weird json problem: %v", testCase.desc, err)
+		}
+		fakeHandler := util.FakeHandler{
+			StatusCode:   200,
+			ResponseBody: string(data),
+		}
+		testServer := httptest.NewServer(&fakeHandler)
+		defer testServer.Close()
+		ch := make(chan interface{}, 1)
+		c := sourceURL{testServer.URL, ch, nil}
+		if err := c.extractFromURL(); err != nil {
+			t.Errorf("%s: Unexpected error: %v", testCase.desc, err)
+			continue
+		}
+		update := (<-ch).(kubelet.PodUpdate)
+
+		for i := range update.Pods {
```

Comment:
I believe we used default because ContainerManifest had no way to specify a namespace. This would be the first static source that allows us to parse Pod instead of ContainerManifest so we can start changing it now. +1 to doing the same on the file.go PR.